src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java

```diff
@@ -159,19 +155,21 @@ protected void configure(final HttpSecurity http) throws Exception {
             .csrf()
                 // Needed for SAML to work properly
                 .disable()
-            .authorizeRequests()
+            .authorizeHttpRequests(auth -> auth
                 // The request to the api is permitted and checked directly
                 // This allows returning a 401 if the user is not logged in via SAML2
                 // to notify the client that a login is needed.
                 .antMatchers("/api/saml2").permitAll()
                 // Every other request must be authenticated. Any request triggers a SAML2
                 // authentication flow
                 .anyRequest().authenticated()
-            .and()
+            )
             // Processes the RelyingPartyRegistrationRepository Bean and installs the filters for SAML2
             .saml2Login()
                 // Redirect back to the root
                 .defaultSuccessUrl("/", true);
```

Comment:
This seems like one indentation too much, it should be on the same level as `saml2Login()`. The formatter is deactivated for these lines.
```suggestion
            .saml2Login()
            // Redirect back to the root
            .defaultSuccessUrl("/", true);
```

Reply:
I don’t think so: `.saml2Login()` returns a `Saml2LoginConfigurer` instead of the `HttpSecurity`, so the `.defaultSuccessUrl` refers to that.

The same for the other class. I now updated the formatting so that everything that refers to `HttpSecurity` directly is consistently at indentation level +1, and otherwise gets indented by an additional level.